src/TransportFactoryFactory.php

```diff
@@ -26,6 +26,7 @@ class TransportFactoryFactory
 {
     public function __invoke(string $dsn, ContainerInterface $container): TransportFactoryInterface
     {
+        /** @psalm-suppress PossiblyUndefinedArrayOffset */
```

Comment:
Ideally, this would go in the baseline, or preferably an inline assertion, for example:

```suggestion
        $parts = explode(':", $dsn, 2);
        assert(count($parts) === 2);
        [$scheme, $config] = $parts;
```

I'll leave it as-is for now because the deprecations will need fixing here too.